yarn-project/p2p/src/client/p2p_client.ts

```diff
@@ -166,12 +167,15 @@ export interface P2P extends P2PApi {
 
   /** Identifies a p2p client. */
   isP2PClient(): true;
-}
+};
 
 /**
  * The P2P client implementation.
  */
-export class P2PClient extends WithTracer implements P2P {
+export class P2PClient<T extends P2PClientType>
+  extends WithTracer
+  implements P2P<P2PClientType.Full>, P2P<P2PClientType.Prover>
```

Comment:
Feels odd to have it implement both interfaces. I'd expect a P2PClient<Prover> to just not have the method `getAttestationsForSlot`, so it doesn't get accidentally called and return empty. But given you're passing it around disguised as a `P2P`, it's fine.